src/relations/db.py

```diff
@@ -237,11 +240,7 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
         user = self._generate_username(join_event)
 
         if self.charm.unit.is_leader():
-            if not (password := self.charm.get_secret(APP_SCOPE, user)):
-                password = pgb.generate_password()
-                self.charm.peers.add_user(user, password)
-        else:
-            password = self.charm.get_secret(APP_SCOPE, user)
```

Comment:
@delgod suggested removing the arbitrary secrets. The charm seems to function mostly the same.